pyproject.toml

```diff
@@ -65,7 +65,7 @@ extend-ignore = [
   "I201",
   "I202",
 ] # these errors conflict with black and/or isort
-type_checking_exempt_modules = ["collections.abc"]
+# type_checking_exempt_modules = ["collections.abc"] # REPO-SPECIFIC CONFIG
```

Comment:
why this change? putting this here just means that imports from `collections.abc` don't have to be in a TYPE_CHECKING block (as is already true for imports from `typing`), which means we don't need to put quotes around annotations as often

Reply:
I guess I thought this exemption was added to avoid adding busywork (from converting annotation formats) to `*-superstaq`.

If you think it's good practice to import from `collections.abc` directly, that's good enough reason for me to leave this exemption in place.  My syntax highlighter is certain happier this way :smile:

Reply:
the main point of the type_checking block is to avoid module-level imports which are used for type checking but aren't necessary at runtime (which can be helpful for avoiding cyclic imports, or deferring costly imports until they're necessary (though we tend to avoid lazy imports so this application isn't particularly relevant to us))

it doesn't really make sense to e.g. put imports from `typing` in a type_checking block, because we'll have to load the `typing` module anyway just to import `TYPE_CHECKING` itself (and for this reason `flake8-type-checking` doesn't require imports from typing to be in a type_checking block). Similarly `collections.abc` is imported within `typing` anyway, so imo it doesn't make sense to put it in a type_checking block either (especially given its ubiquity and how much clumsier it makes the import syntax)

...so tldr imo `flake8-type-checking` is wrong to not exempt collections.abc by default

Reply:
Cool makes total sense